pkg/rbac/postgres/postgres.go

```diff
@@ -0,0 +1,289 @@
+	_ "github.com/jackc/pgx/stdlib"
+	"github.com/omc-college/management-system/pkg/rbac/models"
+)
+
+// DB wraps sql.DB to avoid importing "database/sql" into other packages. closeFn is needed to avoid writing db.db.Close()
+type DB struct {
+	db      *sql.DB
+	closeFn func() error
+}
+
+// NewDB wraps sql.Open() to return *postgres.DB instead of *sql.DB.
+func NewDB(dsn string) (*DB, error) {
+	db, err := sql.Open("pgx", dsn)
+
+	return &DB{
+		db:      db,
+		closeFn: db.Close,
+	}, err
+}
+
+// Wraps db.closeFn()
+func (db *DB) Close() error {
+	return db.closeFn()
+}
+
+func GetAllFeaturesFromDB(db *DB) []models.FeatureEntry {
+	var features []models.FeatureEntry
+	var feature models.FeatureEntry
+
+	query := `SELECT id, name FROM features`
+
+	rows, err := db.db.Query(query)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for rows.Next() {
+		err := rows.Scan(&feature.ID, &feature.Name)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		features = append(features, feature)
+	}
+
+	return features
+}
+
+func GetAllRolesFromDB(db *DB) []models.Role {
+	var roles []models.Role
+	var role models.Role
+
+	query := `SELECT id, name FROM roles`
+
+	rows, err := db.db.Query(query)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for rows.Next() {
+		err := rows.Scan(&role.ID, &role.Name)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		roles = append(roles, role)
+	}
+
+	return roles
+}
+
+func GetRoleFromDB(db *DB, id string) models.Role {
+	var role models.Role
+
+	query := `SELECT id, name FROM roles WHERE roles.id = $1`
+
+	rows, err := db.db.Query(query, id)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for rows.Next() {
+		err := rows.Scan(&role.ID, &role.Name)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	return role
+}
+
+func ConnectFeaturesToEndpoints(db *DB, features []models.FeatureEntry) []models.FeatureEntry {
+	for index, feature := range features {
+		query := `SELECT endpoints.id, endpoints.path, endpoints.method
+				FROM "featuresToEndpoints" LEFT JOIN endpoints ON "featuresToEndpoints"."endpointId" = endpoints.id
+				WHERE "featureId" = $1`
+
+		rows, err := db.db.Query(query, feature.ID)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		var endpoint models.Endpoint
+
+		for rows.Next() {
+			err := rows.Scan(&endpoint.ID, &endpoint.Method, &endpoint.Path)
+			if err != nil {
+				panic(err.Error())
+			}
+
+			features[index].Endpoints = append(features[index].Endpoints, endpoint)
+		}
+	}
+
+	return features
+}
+
+func ConnectRolesToFeatures(db *DB, roles []models.Role, features []models.FeatureEntry) []models.Role {
+	// Copy features slice for every role
+	for index := range roles {
+		roles[index].Entries = make([]models.FeatureEntry, len(features))
+		copy(roles[index].Entries, features)
+	}
+
+	// Define every feature's isAccessGranted for all roles
+	for roleIndex, role := range roles {
+		var currentFeatureId string
+		query := `SELECT "featureId" FROM "rolesToFeatures" WHERE "roleId" = ` + role.ID
```

Comment:
do not concatenate sql queries and variables. Please try to use `github.com/jmoiron/sqlx` package.